dbms/src/Storages/KVStore/MultiRaft/RegionsRangeIndex.cpp

```diff
@@ -171,4 +171,17 @@ RegionsRangeIndex::RootMap::iterator RegionsRangeIndex::split(const TiKVRangeKey
         return do_split(begin_it, new_start);
 }
 
+
+bool RegionsRangeIndex::isRangeOverlapped(const RegionRangeKeys::RegionRange & a, const RegionRangeKeys::RegionRange & b) {
+    auto start = a.first.compare(b.first);
+    if(start == 0) {
+        return true;
+    } else if (start < 0) {
+        return a.second.compare(b.second) >= 0;
+    } else {
+        return b.second.compare(a.second) >= 0;
+    }
```

Comment:
```suggestion
    } else if (start < 0) {
        return a.second.compare(b.first) >= 0;
    } else {
        return b.second.compare(a.first) >= 0;
    }
```

```
        //   |----------|
        // a.first      a.second
        //            |-----------|
        //       b.first        b.second


        //           |----------|
        //       a.first      a.second
        //    |-----------|
        // b.first        b.second
```

Reply:
Add ut about the function `RegionsRangeIndex::isRangeOverlapped`

Reply:
Fixed